src/components/menu/Menu.tsx

```diff
@@ -202,6 +204,12 @@ class Menu extends React.Component<MenuProps> {
     };
 
     handleClick = (event: React.MouseEvent<HTMLUListElement, MouseEvent>) => {
+        const { shouldStopPropagationOnClick } = this.props;
+
+        if (shouldStopPropagationOnClick) {
+            event.stopPropagation();
```

Comment:
Can this handled in onClose that gets passed into the props and called on L202?

Reply:
From my understanding and testing it has to be inside of an onClick function or else it won't stop bubbled up onClick functions from being called

Reply:
Based on what @tjuanitas said here and in my other PR I think it might make most sense to just wrap the components in divs/spans in EUA and handle it there

Reply:
I think that would be best case and leave it up to the consumer to stopPropagation